src/classes/ALLO_AllocationsUtil.cls

```diff
@@ -0,0 +1,1939 @@
+                    PROCESS_DEF.OPP_ALLOC_UPD =>
+                        new Set<PROCESS_DEF>    {PROCESS_DEF.PMT_ALLOC_INITIALIZE,
+                                                PROCESS_DEF.PMT_ALLOC_SYNC_TO_OPP_ALLOC,
+                                                PROCESS_DEF.OPP_ALLOC_SYNC_FINALIZED,
+                                                PROCESS_DEF.OPP_ALLOC_SYNC_NONFINALIZED},
+                    PROCESS_DEF.PMT_ALLOC =>
+                        new Set<PROCESS_DEF>    {PROCESS_DEF.ALL},
+                    PROCESS_DEF.PMT_ALLOC_RECALC =>
+                        new Set<PROCESS_DEF>    {PROCESS_DEF.PMT_ALLOC_SYNC_TO_OPP_ALLOC,
+                                                PROCESS_DEF.OPP_ALLOC_SYNC_FINALIZED,
+                                                PROCESS_DEF.OPP_ALLOC_SYNC_NONFINALIZED},
+                    PROCESS_DEF.PMT_ALLOC_INITIALIZE =>
+                        new Set<PROCESS_DEF>    {PROCESS_DEF.PMT_ALLOC_SYNC_TO_OPP_ALLOC,
+                                                PROCESS_DEF.OPP_ALLOC_SYNC_FINALIZED,
+                                                PROCESS_DEF.OPP_ALLOC_SYNC_NONFINALIZED},
+                    PROCESS_DEF.PMT_ALLOC_SYNC_TO_OPP_ALLOC =>
+                        new Set<PROCESS_DEF>    {PROCESS_DEF.OPP_ALLOC_SYNC_FINALIZED,
+                                                PROCESS_DEF.OPP_ALLOC_SYNC_NONFINALIZED}
+                };
+            }
+
+            return PROCESS_EXCLUSIONS;
+        }
+
+        private set;
+    }
+
+    /** @description Caches excluded Opportunity Recordtype Ids */
+    public static Set<Id> EXCLUDED_OPPORTUNITY_RECORDTYPES {
+        public get {
+            if (EXCLUDED_OPPORTUNITY_RECORDTYPES == null) {
+                EXCLUDED_OPPORTUNITY_RECORDTYPES = (String.isNotBlank(ALLOCATION_SETTINGS.Excluded_Opp_RecTypes__c)) ? new Set<Id>((List<Id>)ALLOCATION_SETTINGS.Excluded_Opp_RecTypes__c.split(';')) : new Set<Id>();
+            }
+            return EXCLUDED_OPPORTUNITY_RECORDTYPES;
+        }
+        private set;
+    }
+
+    /** @description Caches excluded Opportunity Type values */
+    public static Set<String> EXCLUDED_OPPORTUNITY_TYPES {
+        public get {
+            if (EXCLUDED_OPPORTUNITY_TYPES == null) {
+                EXCLUDED_OPPORTUNITY_TYPES = (String.isNotBlank(ALLOCATION_SETTINGS.Excluded_Opp_Types__c)) ? new Set<String>(ALLOCATION_SETTINGS.Excluded_Opp_Types__c.split(';')) : new Set<String>();
+            }
+            return EXCLUDED_OPPORTUNITY_TYPES;
+        }
+        private set;
+    }
+
+    /*******************************************************************************************************
+     * @description Inner class to help Manages Record Exclusions for downstream processes, so it can
+     * process as much as possible while skipping what it has to
+     ********************************************************************************************************/
```

Comment:
NPSP Generally doesn't use full rows of asterisks in ApexDoc header comments. This might be better as
```suggestion
    /**
     * @description Inner class to help Manages Record Exclusions for downstream processes, so it can
     * process as much as possible while skipping what it has to
     */
```